### ISSUE TITLE
Unable to find boost_python3 library

### ISSUE THREAD

**Issue by iqnaul** [osm-fr/osmose-backend#1858]
Hello, I am trying to install osmose-backend using docker in my macos. However it returned this following error:

```
=> ERROR [ 6/11] RUN cd /opt/osmose-backend/modules/osm_pbf_parser && ma  0.4s

 > [ 6/11] RUN cd /opt/osmose-backend/modules/osm_pbf_parser && make:
#0 0.385 Cannot find boost_python3 library from boost_python-py39 or boost_python39
#0 0.385 make: *** [Makefile:18: osm_pbf_parser.cpython-39-aarch64-linux-gnu.so] Error 1

failed to solve: process "/bin/sh -c cd /opt/osmose-backend/modules/osm_pbf_parser && make" did not complete successfully: exit code: 2
```

I am using docker 23.0.5 and docker-compose 2.17.3

Any suggestion to solve this?

Thanks.

**Comment by frodrigo** [comment]
Have you tested a build from scrach ?
```
docker-compose build --no-cache
```

Your macos is running M1 or M2 chip ? Not sure native code could be build on that.

Check of you could build https://github.com/osm-fr/osmose-backend/tree/dev/modules/osm_pbf_parser or https://github.com/hove-io/libosmpbfreader

**Comment by iqnaul** [comment]
hi @frodrigo 
Did try to build from scratch and still return the same error. Yes my macos is running M1 chip, so seems the native code has not support the M1/Docker for M1 chip yeah?

Okay I will try to build the libosmpbfreader directly.

**Comment by pixunil** [comment]
I ran into a similar problem on a Linux host (same error message). However, a compilation on the host machine went fine.
During my troubleshoot, I found that `/usr/lib/x86_64-linux-gnu` (in the docker container) contained `libboost_python311.so` whereas the used python version in the Dockerfile is `python:3.9`. So during the `make`, `libboost_python39.so` is referenced, which did not exist in the container.
I suspect upgrading the python version in the Dockerfile to `python:3.11` has solved my issue, I don’t know whether this solves the original issue.
